File: src/ELibrary.Frontend/src/app/modules/shared/domain/dtos/shop/bookstock/stockBookOrderResponse.ts
import { ClientResponse, mapClientResponseToClient, mapStockBookChangeResponseToStockBookChange, StockBookChangeResponse, StockBookOrder, StockBookOrderType } from "../../../..";

export interface StockBookOrderResponse {
    id: number,
    createdAt: Date,
    updatedAt: Date,
    type: StockBookOrderType,
    totalChangeAmount: number,
    client: ClientResponse,
    stockBookChanges: StockBookChangeResponse[]
}

export function mapStockBookOrderResponseToStockBookOrder(response: StockBookOrderResponse): StockBookOrder {
    return {
        id: response.id,
        createdAt: response.createdAt,
        updatedAt: response.updatedAt,
        type: response.type,
        totalChangeAmount: response.totalChangeAmount,
        client: mapClientResponseToClient(response.client),
        stockBookChanges: response.stockBookChanges.map(x => mapStockBookChangeResponseToStockBookChange(x)),
    }
}

File: src/ELibrary.Backend/LibraryShopEntities/Domain/Dtos/Shop/StockBookOrderResponse.cs
using LibraryShopEntities.Domain.Entities.Shop;

namespace LibraryShopEntities.Domain.Dtos.Shop
{
    public class StockBookOrderResponse
    {
        public int Id { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
        public int TotalChangeAmount { get; set; }
        public StockBookOrderType Type { get; set; }
        public ClientResponse? Client { get; set; }
        public List<StockBookChangeResponse>? StockBookChanges { get; set; }
    }
}
